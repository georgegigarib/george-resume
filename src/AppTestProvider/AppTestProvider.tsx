import React from 'react'
import { I18nextProvider } from 'react-i18next'
import i18n from '@/locales/i18n.config'
import store from '@/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import { vi } from 'vitest'

vi.mock('swiper/react', () => ({
  Swiper: ({
    children,
    onSlideChange,
  }: {
    children: React.ReactNode
    onSlideChange?: (params: { realIndex: number }) => void
  }) => {
    onSlideChange?.({ realIndex: 0 })
    return <div data-testid="swiper">{children}</div>
  },
  SwiperSlide: ({ children }: { children: React.ReactNode }) => <div data-testid="swiper-slide">{children}</div>,
}))

const AppTestProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={children} />
          </Routes>
        </Router>
      </Provider>
    </I18nextProvider>
  )
}

export default AppTestProvider
