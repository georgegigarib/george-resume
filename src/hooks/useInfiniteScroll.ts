import { useEffect, useRef } from 'react'

const useInfiniteScroll = (containerId: string) => {
  const containerRef = useRef<HTMLDivElement>(null)

  useEffect(() => {
    const container = containerRef.current
    if (!container) return

    let animationFrame: number
    let scrollPosition = 0

    try {
      // Duplicamos el contenido para permitir un scroll infinito
      const originalContent = container.innerHTML
      container.innerHTML += originalContent

      // Calculamos el ancho total para ajustar el desplazamiento
      const scrollWidth = container.scrollWidth / 2

      const scrollStep = () => {
        scrollPosition += 0.5 // Ajusta la velocidad del desplazamiento aquÃ­
        if (scrollPosition >= scrollWidth) {
          scrollPosition = 0
        }
        container.scrollLeft = scrollPosition
        animationFrame = requestAnimationFrame(scrollStep)
      }

      scrollStep()
    } catch (error) {
      console.error('Error in useInfiniteScroll:', error)
    }

    // Limpieza al desmontar
    return () => {
      if (animationFrame) cancelAnimationFrame(animationFrame)
    }
  }, [containerId])

  return containerRef
}

export default useInfiniteScroll
