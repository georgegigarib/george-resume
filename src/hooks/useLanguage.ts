import { useTranslation } from 'react-i18next';

export const useLanguage = () => {
  const { i18n } = useTranslation();

  const changeLanguage = (lng: string) => {
    i18n.changeLanguage(lng);
    localStorage.setItem('language', lng);
  };

  const loadLanguageFromStorage = () => {
    const savedLanguage = localStorage.getItem('language') || 'en';
    i18n.changeLanguage(savedLanguage);
  };

  return { changeLanguage, loadLanguageFromStorage };
};
