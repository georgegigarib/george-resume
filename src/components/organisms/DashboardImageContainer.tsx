import React, { useState, useEffect } from "react";
import { useIsMobile } from '@/hooks/useIsMobile';

interface BlurredImageProps {
  imagePath: string;
}

const BlurredImage: React.FC<BlurredImageProps> = ({ imagePath }) => {
  const isMobile = useIsMobile(768);
  const [isLoaded, setIsLoaded] = useState(false);
  const [dimensions, setDimensions] = useState({
    height: '600px',
    boxShadow: `inset 0px 0px 30px 40px black`,
  });

  useEffect(() => {
    const isDarkMode = document.documentElement.classList.contains('dark');
    const blurColor = isDarkMode ? 'black' : 'white';

    if (isMobile) {
      setDimensions({ height: '260px', boxShadow: `inset 0px 0px 18px 20px ${blurColor}` });
    } else {
      setDimensions({ height: '600px', boxShadow: `inset 0px 0px 30px 40px ${blurColor}` });
    }
  }, [isMobile]);

  useEffect(() => {
    setIsLoaded(true);
  }, []);

  return (
    <div className="z-[4]">
      <div
        style={{
          height: dimensions.height,
          boxShadow: dimensions.boxShadow,
          position: "relative",
        }}
      >
        <img
          src={imagePath}
          alt=""
          data-testid="image-test-id"
          style={{
            height: dimensions.height,
            aspectRatio: '16:9',
            position: "relative",
            zIndex: -1,
            opacity: isLoaded ? 1 : 0,
            transition: "opacity 1s ease-in",
          }}
        />
      </div>
    </div>
  );
};

export default BlurredImage;
