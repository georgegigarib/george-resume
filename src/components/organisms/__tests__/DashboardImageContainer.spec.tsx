import { render, screen } from '@testing-library/react'
import { describe, expect, it, vi, beforeEach, Mock } from 'vitest'
import DashboardImageContainer from '@/components/organisms/DashboardImageContainer'
import AppTestProvider from '@/AppTestProvider/AppTestProvider'
import { useIsMobile } from '@/hooks/useIsMobile'
import { useDispatch, useSelector } from 'react-redux'

vi.mock('@/hooks/useIsMobile', () => ({
  useIsMobile: vi.fn(),
}))

vi.mock('react-redux', async () => {
  const actual = await vi.importActual('react-redux')
  return {
    ...actual,
    useSelector: vi.fn(),
    useDispatch: vi.fn(),
  }
})

describe('DashboardImageContainer component', () => {
  const mockDispatch = vi.fn()
  beforeEach(() => {
    vi.clearAllMocks()

    vi.clearAllMocks()
    ;(useDispatch as unknown as Mock).mockReturnValue(mockDispatch)
  })

  const renderComponent = (isMobile: boolean, isDarkMode: boolean, imagePath: string) => {
    ;(useIsMobile as Mock).mockReturnValue(isMobile)
    ;(useSelector as unknown as Mock).mockReturnValue(isDarkMode)
    return render(
      <AppTestProvider>
        <DashboardImageContainer imagePath={imagePath} />
      </AppTestProvider>
    )
  }

  it('matches the snapshot for desktop view in light mode', () => {
    const { asFragment } = renderComponent(false, false, 'path/to/image.jpg')
    expect(asFragment()).toMatchSnapshot()
  })

  it('matches the snapshot for mobile view in light mode', () => {
    const { asFragment } = renderComponent(true, false, 'path/to/image.jpg')
    expect(asFragment()).toMatchSnapshot()
  })

  it('matches the snapshot for desktop view in dark mode', () => {
    const { asFragment } = renderComponent(false, true, 'path/to/image.jpg')
    expect(asFragment()).toMatchSnapshot()
  })

  it('matches the snapshot for mobile view in dark mode', () => {
    const { asFragment } = renderComponent(true, true, 'path/to/image.jpg')
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders the image with correct src', async () => {
    renderComponent(false, false, 'path/to/image.jpg')
    const imgElement = screen.getByTestId('image-test-id')
    expect(imgElement).toHaveAttribute('src', 'path/to/image.jpg')
  })

  it.skip('applies the correct styles based on mobile and dark mode', async () => {
    renderComponent(true, true, 'path/to/image.jpg')

    expect(await screen.getByTestId('image-test-id')).toHaveStyle({
      height: '260px',
      boxShadow: 'inset 0px 0px 18px 20px #000',
    })
  })

  it.skip('applies the correct styles based on desktop and light mode', async () => {
    renderComponent(false, false, 'path/to/image.jpg')
    const imgElement = await screen.getByTestId('image-test-id')

    expect(imgElement).toHaveStyle({
      height: '600px',
      boxShadow: 'inset 0px 0px 40px 40px white',
    })
  })
})
