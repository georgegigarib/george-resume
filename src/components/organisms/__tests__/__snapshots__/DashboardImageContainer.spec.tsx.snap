// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DashboardImageContainer component > matches the snapshot for desktop view in dark mode 1`] = `
<DocumentFragment>
  <div
    class="z-[4]"
  >
    <div
      data-testid="image-container-testid"
      style="height: 600px; box-shadow: inset 0px 0px 40px 40px #000; position: relative;"
    >
      <div
        class="absolute inset-0 flex items-center justify-center"
      >
        <span
          class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1yonwsg-MuiCircularProgress-root"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </span>
      </div>
      <img
        alt=""
        class="undefined opacity-0"
        data-testid="image-testid"
        src="path/to/image.jpg"
        style="height: 600px; aspect-ratio: 16:9; position: relative; z-index: -1; opacity: 1; transition: opacity 1s ease-in;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`DashboardImageContainer component > matches the snapshot for desktop view in light mode 1`] = `
<DocumentFragment>
  <div
    class="z-[4]"
  >
    <div
      data-testid="image-container-testid"
      style="height: 600px; box-shadow: inset 0px 0px 40px 40px #fff; position: relative;"
    >
      <div
        class="absolute inset-0 flex items-center justify-center"
      >
        <span
          class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1yonwsg-MuiCircularProgress-root"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </span>
      </div>
      <img
        alt=""
        class="undefined opacity-0"
        data-testid="image-testid"
        src="path/to/image.jpg"
        style="height: 600px; aspect-ratio: 16:9; position: relative; z-index: -1; opacity: 1; transition: opacity 1s ease-in;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`DashboardImageContainer component > matches the snapshot for mobile view in dark mode 1`] = `
<DocumentFragment>
  <div
    class="z-[4]"
  >
    <div
      data-testid="image-container-testid"
      style="height: 260px; box-shadow: inset 0px 0px 18px 20px #000; position: relative;"
    >
      <div
        class="absolute inset-0 flex items-center justify-center"
      >
        <span
          class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1yonwsg-MuiCircularProgress-root"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </span>
      </div>
      <img
        alt=""
        class="undefined opacity-0"
        data-testid="image-testid"
        src="path/to/image.jpg"
        style="height: 260px; aspect-ratio: 16:9; position: relative; z-index: -1; opacity: 1; transition: opacity 1s ease-in;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`DashboardImageContainer component > matches the snapshot for mobile view in light mode 1`] = `
<DocumentFragment>
  <div
    class="z-[4]"
  >
    <div
      data-testid="image-container-testid"
      style="height: 260px; box-shadow: inset 0px 0px 18px 20px #fff; position: relative;"
    >
      <div
        class="absolute inset-0 flex items-center justify-center"
      >
        <span
          class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1yonwsg-MuiCircularProgress-root"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-1a8epmm-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </span>
      </div>
      <img
        alt=""
        class="undefined opacity-0"
        data-testid="image-testid"
        src="path/to/image.jpg"
        style="height: 260px; aspect-ratio: 16:9; position: relative; z-index: -1; opacity: 1; transition: opacity 1s ease-in;"
      />
    </div>
  </div>
</DocumentFragment>
`;
